creme/linear_model/adpredictor.py

```diff
@@ -0,0 +1,175 @@
+import collections
+import numbers
+import numpy as np 
+from scipy import stats
+from .. import base 
+from .. import optim
+from .. import utils
+
+
+
+__all__ = [
+    'AdPredictor'
+]
+
+
+class NormalPrior(optim.initializers.Initializer):
```

Comment:
I'm not sure this has inherit from `Initializer`. Initializers are here to provide a way to generate random weights when initializing a model, but your case is a bit different.

I understand that there is some need for custom weights, in which case you could use something like a `collections.namedtuple`. You could also store the means and variances in two separate lists.

There's some other style stuff to change, but I'll add some comments once this is refactored :)

Reply:
Ok I will try that. Thanks for all your comments, it helps a lot !